src/hgraph/_wiring/_decorators.py

```diff
@@ -551,42 +553,76 @@ def register_service(path: str, implementation, resolution_dict=None, **kwargs):
     """
     from hgraph._wiring._wiring_node_class._wiring_node_class import PreResolvedWiringNodeWrapper
     from hgraph._wiring._wiring_node_class._service_impl_node_class import ServiceImplNodeClass
+    from hgraph._wiring._wiring_node_class._adaptor_impl_node_class import AdaptorImplNodeClass
 
     if isinstance(implementation, PreResolvedWiringNodeWrapper):
         implementation = implementation.underlying_node
         resolution_dict = implementation.resolved_types or {}
     else:
         resolution_dict = implementation._convert_item(resolution_dict) if resolution_dict else {}
 
-    if not isinstance(implementation, ServiceImplNodeClass):
+    if not isinstance(implementation, (ServiceImplNodeClass, AdaptorImplNodeClass)):
         raise CustomMessageWiringError("The provided implementation is not a 'service_impl' wrapped function.")
 
     from hgraph import WiringGraphContext
 
     for i in implementation.interfaces:
-        WiringGraphContext.instance().register_service_impl(
-            i, i.full_path(path), implementation, kwargs, resolution_dict
-        )
+        WiringGraphContext.instance().register_service_impl(i, path, implementation, kwargs, resolution_dict)
 
 
-def service_adaptor(interface):
+def adaptor(interface, resolvers: Mapping[TypeVar, Callable] = None):
```

Comment:
Explain how this different from service interfaces / implementations.
Note what the expected usage pattern is, i.e. what happens if there are multiple instances of adaptor created?